frontend/packages/dev-console/src/components/builds/BuildsTabListPage.tsx

```diff
@@ -62,6 +73,27 @@ const BuildsTabListPage: React.FC = () => {
   const SHIPWRIGHT_BUILD_V1ALPHA1 = useFlag('SHIPWRIGHT_BUILD_V1ALPHA1');
 
   const shipwrightBuildEnabled = SHIPWRIGHT_BUILD || SHIPWRIGHT_BUILD_V1ALPHA1;
+
+  /* Redirect to last visited tab */
+  React.useEffect(() => {
+    if (preferredTabLoaded && namespace) {
+      if (preferredTab === 'shipwright-builds' && shipwrightBuildEnabled) {
+        navigate(`/builds/ns/${namespace}/shipwright-builds`, { replace: true });
+      } else {
+        navigate(`/builds/ns/${namespace}`, { replace: true });
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [namespace, preferredTabLoaded]);
+
+  React.useEffect(() => {
+    // update the preferred tab
+    if (preferredTabLoaded && namespace) {
+      setPreferredTab(currentTab === 'shipwright-builds' ? 'shipwright-builds' : 'buildconfigs');
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [namespace, currentTab, preferredTabLoaded]);
```

Comment:
good catches, but for the first hook, i think that `preferredTab` should not be in the array as we only want to navigate the user once.

i am also leaning towards no for `shipwrightbuildenabled` as well because i don't want to redirect the user when shipwright is enabled, only on first open after everything is loaded